mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -1079,7 +1086,28 @@ impl MakerSwap {
             ]));
         }
 
+        let taker_payment = self.r().taker_payment.clone();
+        if let Some(payment) = &taker_payment {
+            if let Err(e) = self.taker_coin.on_maker_payment_refund_start(&payment.tx_hex).await {
```

Comment:
This is really confusing because `TakerSwapOps::on_maker_payment_refund_start` is called on the Maker side if **only** the taker payment has been sent.
Shouldn't `TakerSwapOps` be used on the Taker side only?

Also I'm really not sure if `MakerSwapOps` and `TakerSwapOps` names are suitable as the coins traits.
What if we have `MakerSwapMakerCoin`, `MakerSwapTakerCoin`, `TakerSwapMakerCoin`, `TakerSwapTakerCoin`.

Reply:
Yeah, this looks more clear than `Maker/TakerSwapOps`, so I agree 🙂

Reply:
> Will open an issue for the refactor/spliting of SwapOps https://github.com/KomodoPlatform/atomicDEX-API/pull/1592#discussion_r1063375403 to the 4 above traits.

https://github.com/KomodoPlatform/atomicDEX-API/issues/1610

Reply:
Consider this as solved!